src/abi/map.ts

```diff
@@ -500,4 +500,81 @@ export const MapFunctions: Map<string, AbiFunction> = new Map([
             },
         },
     ],
+    [
+        "exists",
+        {
+            name: "exists",
+            resolve(ctx, args, ref) {
+                // Check arguments
+                if (args.length !== 2) {
+                    throwCompilationError("exists expects one argument", ref); // Ignore self argument
+                }
+                const self = args[0]!;
+                const key = args[1]!;
+                if (self.kind !== "map") {
+                    throwCompilationError(
+                        "exists expects a map as self argument",
+                        ref,
+                    ); // Should not happen
```

Comment:
Let's add more negative typechecker tests with at least one test like `42.exists` and see if this comment is correct